### PR TITLE
implement flavors

### PR DESCRIPTION
**Pull request by weiwei**
This is to address #101, as well as other frequent xml schema support requirements.

I've thought about `JunitXml(flavor='pytest')`, but typings will be tricky when it comes to dealing with elements like `TestSuite` and `TestCase`, which will also be different with those in the "standard" implementation, For example, autocompletion won't know this `TestCase` now has a `rerun_errors()` method. 

With implementation in this PR, a flavor and its elements are imported and used separately, and autocompletion works on each case.

```python
from junitparser.flavors.pytest_xml import JUnitXml, TestSuite, TestCase, RerunFailure

# Create cases
case1 = TestCase('case1') # A pytest case
case1.group = 'Group1' # pytest specific
case1.add_rerun_result(RerunFailure('Server Error', '500') # pytest specific

# Create suite and add cases
suite = TestSuite('suite1') # A pytest suite
suite.add_property('build', '55')
suite.add_testcase(case1)
suite.remove_testcase(case2)

# Add suite to JunitXml
xml = JUnitXml() # A pytest JUnitXml class
xml.add_testsuite(suite)
xml.write('junit.xml')
```

Also address #97 by adding rudimentary types

### CODE REVIEW

File: junitparser/flavors/pytest_xml.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
+import itertools
+from junitparser.junitparser import JUnitXml, JUnitXmlError, Result, TestSuite as OrigTestSuite, Attr, \
+    SystemOut, SystemErr, TestCase as OrigTestCase, System
+
+
+class PytestXml(JUnitXml):
```
Comment (weiwei) [inline junitparser/flavors/pytest_xml.py:6]: Maybe name it to `JUnitXml` as well. Could result in less cognitive load. Flavors are separated by file names already.

Comment (EnricoMi) [reply junitparser/flavors/pytest_xml.py:6]: Yes, this should be
```suggestion
class JUnitXml(junitparser.JUnitXml):
```
Then user code looks identical, no matter which import they use:
```
from junitparser.junitparser import JUnitXml
from junitparser.flavors.pytest_xml import JUnitXml
```

File: tests/test_pytest_xml.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+import unittest
+from junitparser.flavors.pytest_xml import PytestXml, TestSuite, TestCase, FlakyError, FlakyFailure, RerunError, RerunFailure
```
Comment (EnricoMi) [inline tests/test_pytest_xml.py:4]: I'd prefer
```
from junitparser.pytest_xml import ...
```

File: junitparser/junitparser.py
Comment (weiwei) [inline junitparser/junitparser.py:1]: No code change. Just that orders need to be adjusted for the type hints to work.

File: tests/test_general.py
Comment (weiwei) [inline tests/test_general.py:1]: Format change curtesy of black.

**Review by EnricoMi — COMMENTED** [review]
Looks great!

**Pull request merged**